AlfrescoApp/Model/Managers/RealmSyncHelper.m

```diff
@@ -0,0 +1,137 @@
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ ******************************************************************************/
+
+#import "RealmSyncHelper.h"
+#import "RealmManager.h"
+#import "SyncNodeStatus.h"
+#import "SyncConstants.h"
+#import "AccountManager.h"
+
+@interface RealmSyncHelper()
+
+@property (nonatomic, strong) AlfrescoFileManager *fileManager;
+
+@end
+
+@implementation RealmSyncHelper
+
++ (RealmSyncHelper *)sharedHelper
+{
+    static dispatch_once_t predicate = 0;
+    __strong static id sharedObject = nil;
+    dispatch_once(&predicate, ^{
+        sharedObject = [[self alloc] init];
+    });
+    return sharedObject;
+}
+
+- (instancetype)init
+{
+    self = [super init];
+    if (self)
+    {
+        self.fileManager = [AlfrescoFileManager sharedManager];
+    }
+    return self;
+}
+
+- (NSString *)syncNameForNode:(AlfrescoNode *)node inRealm:(RLMRealm *)realm
+{
+    RealmSyncNodeInfo *nodeInfo = [[RealmManager sharedManager] syncNodeInfoForObjectWithId:[self syncIdentifierForNode:node] inRealm:realm];
+    
+    if (nodeInfo.syncContentPath.length == 0)
+    {
+        NSString *newName = @"";
+        NSString *nodeExtension = [node.name pathExtension];
+        
+        if (nodeExtension.length == 0)
+        {
+            newName = [[self syncIdentifierForNode:node] lastPathComponent];
+        }
+        else
+        {
+            newName = [NSString stringWithFormat:@"%@.%@", [[self syncIdentifierForNode:node] lastPathComponent], nodeExtension];
+        }
+        return newName;
+    }
+    return [nodeInfo.syncContentPath lastPathComponent];
+}
+
+- (NSDate *)lastDownloadedDateForNode:(AlfrescoNode *)node inRealm:(RLMRealm *)realm
+{
+    RealmSyncNodeInfo *nodeInfo = [[RealmManager sharedManager] syncNodeInfoForObjectWithId:[self syncIdentifierForNode:node] inRealm:realm];
+    return nodeInfo.lastDownloadedDate;
+}
+
+- (SyncNodeStatus *)syncNodeStatusObjectForNodeWithId:(NSString *)nodeId inSyncNodesStatus:(NSDictionary *)syncStatuses
+{
+    SyncNodeStatus *nodeStatus = [syncStatuses objectForKey:nodeId];
+    
+    if (!nodeStatus && nodeId)
+    {
+        nodeStatus = [[SyncNodeStatus alloc] initWithNodeId:nodeId];
+        [syncStatuses setValue:nodeStatus forKey:nodeId];
+    }
+    
+    return nodeStatus;
+}
+
+- (NSString *)syncContentDirectoryPathForAccountWithId:(NSString *)accountId
+{
+    NSString *contentDirectory = [self.fileManager syncFolderPath];
+    if (accountId)
+    {
+        contentDirectory = [contentDirectory stringByAppendingPathComponent:accountId];
+    }
+    BOOL isDirectory;
```

Comment:
Why do you need isDirectory var? If you don't need it, the fileExistsAtPath: will be enough.